File: babyfoot/babyfoot/Models/Tournament.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;

namespace babyfoot.Models
{
    public enum TournamentState
    {
        Pool, Semifinal, Final, Ended
    };

    public class Tournament
    {
        [DatabaseGeneratedAttribute(DatabaseGeneratedOption.Identity)]

        public int TournamentId { get; set; }

        public String Token { get; set; }

        public System.DateTime CreateDate { get; set; }



        public TournamentState State { get; set; }

        public ICollection<Match> Matches { get; set; }
    }
}

File: babyfoot/babyfoot/ClientApp/src/app/entities/Tournament.ts
import { Match } from './Match';
import { TournamentService } from '../service/tournament.service';
import { TournamentTeam } from './TournamentTeam';

export interface Tournament
{

    token: string;
    date: Date;
    state: string;
    teams: TournamentTeam[];
    matches: Match[];
}
